src/System.Memory/src/System/Buffers/Text/Utf8Formatter/FormattingHelpers.cs

```diff
@@ -16,111 +16,213 @@ namespace System.Buffers.Text
     // code must have already done all the necessary validation.
     internal static class FormattingHelpers
     {
-        // For the purpose of formatting time, the format specifier contains room for
-        // exactly 7 digits in the fraction portion. See "Round-trip format specifier"
-        // at the following URL for more information.
-        // https://msdn.microsoft.com/en-us/library/az4se3k1(v=vs.110).aspx#Roundtrip
-        private const int FractionDigits = 7;
-
         // A simple lookup table for converting numbers to hex.
-        private const string HexTable = "0123456789abcdef";
+        internal const string HexTableLower = "0123456789abcdef";
 
-        #region UTF-8 Helper methods
+        internal const string HexTableUpper = "0123456789ABCDEF";
 
+        /// <summary>
+        /// Returns the symbol contained within the standard format. If the standard format
+        /// has not been initialized, returns the provided fallback symbol.
+        /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void WriteHexByte(byte value, ref byte buffer, int index)
+        public static char GetSymbolOrDefault(in StandardFormat format, char defaultSymbol)
         {
-            Unsafe.Add(ref buffer, index) = (byte)HexTable[value >> 4];
-            Unsafe.Add(ref buffer, index + 1) = (byte)HexTable[value & 0xF];
+            // This is equivalent to the line below, but it is written in such a way
+            // that the JIT is able to perform more optimizations.
+            //
+            // return (format.IsDefault) ? defaultSymbol : format.Symbol;
+
+            var symbol = format.Symbol;
+            if (symbol == default && format.Precision == default)
+            {
+                symbol = defaultSymbol;
+            }
+            return symbol;
         }
 
+        #region UTF-8 Helper methods
+
+        /// <summary>
+        /// Fills a buffer with the ASCII character '0' (0x30).
+        /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void WriteFractionDigits(long value, int digitCount, ref byte buffer, int index)
+        public static void FillWithAsciiZeros(Span<byte> buffer)
         {
-            for (int i = FractionDigits; i > digitCount; i--)
-                value /= 10;
+            // This is a faster implementation of Span<T>.Fill().
```

Comment:
Why is this faster? Is it due to buffer.Length being too small in this case?

Wouldn't Fill using InitBlockUnaligned be just as fast, if not faster:
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Span.cs#L219-L231